src/inmanta/data/__init__.py

```diff
@@ -1274,11 +1281,31 @@ class Compile(BaseDocument):
 
     @classmethod
     async def get_unhandled_compiles(cls) -> "List[Compile]":
+        """ Get all compiles that have completed but for which listeners have not been notified yet.
+        """
         results = await cls.select_query(
             f"SELECT * FROM {cls.table_name()} WHERE NOT handled and completed IS NOT NULL ORDER BY requested ASC", []
         )
         return results
 
+    @classmethod
+    async def get_next_compiles_for_environment(cls, environment_id: uuid.UUID) -> "List[Compile]":
+        """ Get the queue of compiles that are scheduled in FIFO order.
+        """
+        results = await cls.select_query(
+            f"SELECT * FROM {cls.table_name()} WHERE environment=$1 AND NOT handled and completed IS NULL "
+            "ORDER BY requested ASC",
+            [environment_id],
```

Comment:
the rest of the code uses
```
  [cls._get_value(environment_id)]
```

This may not be strictly required in this case, but it is good for consistency